classes/Address.php

```diff
@@ -442,12 +443,21 @@ public static function getCountryAndState($id_address)
      */
     public static function addressExists($id_address)
     {
-        return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
-            'SELECT `id_address`
-            FROM ' . _DB_PREFIX_ . 'address a
-            WHERE a.`id_address` = ' . (int) $id_address,
-            false
-        );
+        if ($id_address == 0) {
+            return false;
+        }
+        $cacheKey = 'address_exists_' . $id_address;
```

Comment:
Hi @lmeyer1 the PR and the optim is legitimate as is, but as we discussed it would be nice if the method is called only when appropriate.

I searched for its use and they are quite limited, only three references in the cart class:
- https://github.com/PrestaShop/PrestaShop/blob/6f447538fff3535ffc18d4d911441c07059b3e0f/classes/Cart.php#L909
- https://github.com/PrestaShop/PrestaShop/blob/6f447538fff3535ffc18d4d911441c07059b3e0f/classes/Cart.php#L3579
- https://github.com/PrestaShop/PrestaShop/blob/6f447538fff3535ffc18d4d911441c07059b3e0f/classes/Cart.php#L2403

All other references check that the id is not 0 before, so only those two need fixing. If it was a hug work to fix everywhere it would have deserved another dedicated PR. But since it's only three small fixes I think it can be introduced in this PR. What do you think?